packages/amplify-category-api/src/provider-utils/awscloudformation/service-walkthroughs/appSync-rds-db-config.ts

```diff
@@ -0,0 +1,54 @@
+import { prompter, printer, integer } from '@aws-amplify/amplify-prompts';
+import {
+  ImportedDataSourceType,
+  ImportedDataSourceConfig,
+} from '@aws-amplify/graphql-transformer-core';
+import { parseDatabaseUrl } from '../utils/database-url';
+
+/**
+ * Gathers database configuration information
+ * @param engine the database engine
+ * @returns a promise resolving to the database configuration to be used as an AppSync Data Source
+ */
+export const databaseConfigurationInputWalkthrough = async (engine: ImportedDataSourceType): Promise<ImportedDataSourceConfig> => {
+  printer.info('Please provide the following database connection information:');
+  const url = await prompter.input('Enter the database url or host name:');
```

Comment:
we need to have a check in here so that we don't accept empty input (when the user accidentally just hits enter) and re-prompt. You can do so by passing a second argument to this method:
```
{
      validate: minLength(1),
 }
```

Reply:
Thanks for the callout. This wasn't present in the original code so I missed the validation.